### PR TITLE
terraform: run `init` with no-color, too

### PR DESCRIPTION
**Pull request by azrdev**
##### SUMMARY

When running `terraform init` fails, it would output ansi color sequences, making the output hard to read.

Maybe setting TF_IN_AUTOMATION would also be beneficial: https://www.terraform.io/cli/config/environment-variables#tf_in_automation

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

terraform

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! Could you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Thanks.

Also a question: does it make sense to pass that flag only here, and nowhere else? Also, is this flag supported by all versions of terraform?

**Comment by azrdev** [comment]
Current terraform indeed has `-no-color` as a global argument, so I assume it's applicable to all commands: 
https://github.com/hashicorp/terraform/blob/71dec30/internal/command/arguments/view.go#L26

I don't know when this parameter was introduced, https://www.terraform.io/cli/v1.1.x/commands/init and the current 1.2 have it without version notice. S/o would have to look that up.

File: changelogs/5147-terraform-init-no-color.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - terraform - run ``terraform init`` with ``-no-color`` not to mess up the stdout of the task.
```
Comment (felixfontein) [inline changelogs/5147-terraform-init-no-color.yml:2]:
```suggestion
  - terraform - run ``terraform init`` with ``-no-color`` not to mess up the stdout of the task (https://github.com/ansible-collections/community.general/pull/5147).
```
Please note that this file must be in the subdirectory `fragments/` of `chanelogs/`.

**Comment by felixfontein** [comment]
@azrdev ping?

needs_info

**Comment by azrdev** [comment]
@felixfontein changed as proposed

**Review by felixfontein — APPROVED** [review]
If nobody complains, I'll merge in a couple of days.

**Comment by felixfontein** [comment]
@azrdev thanks for your contribution!

**Pull request merged**